frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -138,9 +161,20 @@ const CloudShellTerminal: React.FC<CloudShellTerminalProps> = ({ user, onCancel
     );
   }
 
+  if (admin) {
+    return (
+      <CloudShellAdminSetup
```

Comment:
Seems in addition to admin-specific logic here, we need to modify place where we pick the existing dashboard. To reproduce it just apply with terminal the following and try console
```yaml
apiVersion: workspace.devfile.io/v1alpha1
kind: DevWorkspace
metadata:
  annotations:
    controller.devfile.io/restricted-access: "true"
  labels:
    console.openshift.io/terminal: "true"
    controller.devfile.io/creator: ""
  name: terminal-kasasa
  namespace: default
spec:
  routingClass: web-terminal
  started: true
  template:
    components:
    - plugin:
        id: redhat-developer/web-terminal/4.5.0
        name: web-terminal
```

Simpler admin specific logic for picking existing devworkspace - could be just ignore terminals in different namespace from openshift-terminal.

A bit better but requires some UI - if any found out from openshift-terminal namespace - propose admin to remove it and create in openshift-terminal.

I'm fully fine with simpler option.